src/server/lib/logger.ts

```diff
@@ -1,9 +1,68 @@
+import { AWSError, Kinesis } from 'aws-sdk';
 import { Logger, LogLevel } from '@/server/models/Logger';
 import { createLogger, transports } from 'winston';
+import Transport, { TransportStreamOptions } from 'winston-transport';
 import { formatWithOptions, InspectOptions } from 'util';
+import { awsConfig } from './awsConfig';
+import { getConfiguration } from './getConfiguration';
+
+const {
+  stage,
+  aws: { instanceId, kinesisStreamName },
+} = getConfiguration();
+
+// custom "Winston Transport" to send logs to the AWS kinesis stream
+// see https://github.com/winstonjs/winston-transport#usage
+class KinesisTransport extends Transport {
+  private kinesis: Kinesis;
+
+  constructor(opts?: TransportStreamOptions) {
+    super(opts);
+
+    this.kinesis = new Kinesis(awsConfig);
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  public log(info: any, callback: () => void) {
+    const { level, message } = info;
+
+    setImmediate(() => {
+      this.emit('logger', level);
+      if (kinesisStreamName) {
+        this.kinesis
+          .putRecord({
+            StreamName: kinesisStreamName,
+            PartitionKey: stage,
+            Data: JSON.stringify({
+              type: 'app',
+              app: 'identity-gateway',
+              stack: 'identity',
+              path: '/var/log/identity-gateway.log',
+              instance_id: instanceId,
+              stage,
+              level,
+              message,
+            }),
+          })
+          .promise()
+          .catch((error: AWSError) => {
+            if (error.code === 'ExpiredToken' && stage === 'DEV') {
+              logger.warn(
```

Comment:
does calling log on logger failure cause an infinite loop?

Reply:
good spot! i'll change it to a `console.warn`, and i'll remove the logging failure error